File: OpenBullet2.Web/Dtos/Settings/EnvironmentSettingsDto.cs
using RuriLib.Models.Environment;

namespace OpenBullet2.Web.Dtos.Settings;

/// <summary>
/// DTO that represents the environment settings.
/// </summary>
public class EnvironmentSettingsDto
{
    /// <summary>
    /// The available wordlist types.
    /// </summary>
    public List<WordlistType> WordlistTypes { get; set; } = new();

    /// <summary>
    /// The available custom statuses.
    /// </summary>
    public List<CustomStatus> CustomStatuses { get; set; } = new();

    /// <summary>
    /// The available export formats.
    /// </summary>
    public List<ExportFormat> ExportFormats { get; set; } = new();
}

File: openbullet2-web-client/src/app/main/dtos/settings/environment-settings.dto.ts
export interface EnvironmentSettingsDto {
  wordlistTypes: WordlistType[];
  customStatuses: CustomStatus[];
  exportFormats: ExportFormat[];
}

export interface WordlistType {
  name: string;
  regex: string;
  verify: boolean;
  separator: string;
  slices: string[];
  slicesAlias: string[];
}

export interface CustomStatus {
  name: string;
  color: string;
}

export interface ExportFormat {
  format: string;
}
